### PR TITLE
Feature/add authorization

### PR DESCRIPTION
**Pull request by Leblyad**

### CODE REVIEW

**Comment by Leblyad** [comment]
In this api each user has only one role.

File: AuthorizationAPI/Presentation/Controllers/AuthenticationController.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        private readonly IAccountService _authenticationService;
+
+        public AuthenticationController(IAccountService authenticationService)
+        {
+            _authenticationService = authenticationService;
+        }
+
+        /// <summary>
+        /// Creates a new user
```
Comment (kseniya-kalita) [inline AuthorizationAPI/Presentation/Controllers/AuthenticationController.cs:19]: Nice idea to have comments, like it! Please add also possible returned status cods. And add the possibility to view your comment in swagger.

File: AuthorizationAPI/Application/Serrvices.Abstractions/IAccountService.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using AuthorizationAPI.Core.Entities.Models.AuthorizationDTO;
+
+namespace AuthorizationAPI.Application.Serrvices.Abstractions
```
[the remark this answers is not in the document]
Comment (kseniya-kalita) [reply AuthorizationAPI/Application/Serrvices.Abstractions/IAccountService.cs:3]: And place this folder inside Services

**Review by kseniya-kalita — CHANGES_REQUESTED** [review]
Good work! Please review and resolve comments to have your code better. After that ping me and I'll review it one more time.

File: AuthorizationAPI/Infrastructure/Repository/AuthenticationDbContext.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace AuthorizationAPI.Infrastructure.Repository
+{
+    public class AuthenticationDbContext : IdentityDbContext<User>
+    {
+        public AuthenticationDbContext(DbContextOptions<AuthenticationDbContext> options)
+            : base(options)
+        {
+        }
+
```
[the remark this answers is not in the document]
Comment (Leblyad) [reply AuthorizationAPI/Infrastructure/Repository/AuthenticationDbContext.cs:15]: Sorry, i tried

File: AuthorizationAPI/Presentation/Controllers/AuthenticationController.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            await _authenticationService.CreateUser(userForCreation, role);
+            return Ok();
+        }
+
+        /// <summary>
+        /// Add role to user
+        /// | Required role: Administrator
+        /// </summary>
+        /// <param name="login"></param>
```
[the remark this answers is not in the document]
Comment (Leblyad) [reply AuthorizationAPI/Presentation/Controllers/AuthenticationController.cs:33]: Username, i renamed it.

File: AuthorizationAPI/Application/Services/AccountService.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                    await _userManager.AddToRoleAsync(user, nameof(UserRole.Receptionist));
+                    break;
+                case nameof(UserRole.Pacient):
+                    await _userManager.AddToRoleAsync(user, nameof(UserRole.Pacient));
+                    break;
+                case nameof(UserRole.Doctor):
+                    await _userManager.AddToRoleAsync(user, nameof(UserRole.Doctor));
+                    break;
+                default:
+                    throw new Exception("Role not exists");
```
[the remark this answers is not in the document]
Comment (Leblyad) [reply AuthorizationAPI/Application/Services/AccountService.cs:83]: Now i changed approach.

File: AuthorizationAPI/Core/Entities/Models/AuthorizationDTO/AuthenticatedUserInfo.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿namespace AuthorizationAPI.Core.Entities.Models.AuthorizationDTO
+{
+    public class AuthenticatedUserInfo
+    {
+        public string AccessToken { get; set; }
+
+        public string RefreshToken { get; set; }
+
+        public IList<string> UserRoles { get; set; }
```
[the remark this answers is not in the document]
Comment (Leblyad) [reply AuthorizationAPI/Core/Entities/Models/AuthorizationDTO/AuthenticatedUserInfo.cs:9]: No, just one. Resolve it.

File: AuthorizationAPI/Presentation/Controllers/AuthenticationController.cs (hunk 1 above)
Comment (Leblyad) [reply AuthorizationAPI/Presentation/Controllers/AuthenticationController.cs:19]: Done.

File: InnoClinic.AuthorizationAPI/appsettings.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft.AspNetCore": "Warning"
+    }
+  },
+  "AllowedHosts": "*",
+  "applicationUrl": "https://localhost:7141",
+  "ConnectionStrings": {
+    "sqlConnection": "server=.\\SQLEXPRESS; database=Authorization; Integrated Security=true"
```
Comment (kseniya-kalita) [inline InnoClinic.AuthorizationAPI/appsettings.json:11]: One last point: please add 'Db' ending to your database name :)

**Review by kseniya-kalita — APPROVED** [review]
Great job! Please resolve these small changes and merge without resending this PR to me

**Pull request merged**